<?xml version="1.0" ?>
<robot name="Design" xmlns:xacro="http://www.ros.org/wiki/xacro">

<!--<xacro:include filename="$(find version1_description)/urdf/materials.xacro" />
<xacro:include filename="$(find version1_description)/urdf/Design.gazebo" /> -->

<xacro:include filename="$(find main_description)/urdf/Design.trans" />


<!--  PROPERTIES  -->

<xacro:property name="strength" value="1000"/>
<xacro:property name="speed" value = "0.01"/>
<xacro:property name="damping" value = "0"/>

<xacro:property name="offset" value="-0.05"/>

<!-- LINKS  -->

<!--<xacro:include filename="$(find design2_description)/urdf/materials.xacro" />-->
<!--<xacro:include filename="$(find design2_description)/urdf/design2.trans" />-->
<!--<xacro:include filename="$(find design2_description)/urdf/design2.gazebo" />-->

<link name="base_link">
  <inertial>
    <origin rpy="0 0 0" xyz="-3.054172551678278e-15 2.5451437930652316e-17 0.009826233848161996"/>
    <mass value="2.1038693653686527"/>
    <inertia ixx="0.002498" ixy="0.0" ixz="-0.0" iyy="0.002498" iyz="0.0" izz="0.004751"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <geometry>
      <mesh filename="package://main_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <geometry>
      <mesh filename="package://main_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="Link1_1">
  <inertial>
    <origin rpy="0 0 0" xyz="-6.1328821587432384e-06 0.007231132282278709 0.010400791755463774"/>
    <mass value="0.8688143187550561"/>
    <inertia ixx="0.000613" ixy="0.0" ixz="0.0" iyy="0.000486" iyz="-8.3e-05" izz="0.000932"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-0.0 -0.0 -0.025"/>
    <geometry>
      <mesh filename="package://main_description/meshes/Link1_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-0.0 -0.0 -0.025"/>
    <geometry>
      <mesh filename="package://main_description/meshes/Link1_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="Link2_1">
  <inertial>
    <origin rpy="0 0 0" xyz="-1.4589717024326475e-07 0.009220625083884316 0.09374198522195988"/>
    <mass value="0.6628237762897414"/>
    <inertia ixx="0.002038" ixy="0.0" ixz="0.0" iyy="0.001955" iyz="-0.000356" izz="0.000168"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="2.1e-05 -0.004903 -0.05"/>
    <geometry>
      <mesh filename="package://main_description/meshes/Link2_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="2.1e-05 -0.004903 -0.05"/>
    <geometry>
      <mesh filename="package://main_description/meshes/Link2_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="Link3_1">
  <inertial>
    <origin rpy="0 0 0" xyz="1.9618427109714154e-05 0.006527757535011174 0.07498286609368165"/>
    <mass value="0.4910812167638009"/>
    <inertia ixx="0.001039" ixy="-0.0" ixz="-0.0" iyy="0.001007" iyz="-0.000178" izz="8.5e-05"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="2.1e-05 -0.004903 -0.2"/>
    <geometry>
      <mesh filename="package://main_description/meshes/Link3_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="2.1e-05 -0.004903 -0.2"/>
    <geometry>
      <mesh filename="package://main_description/meshes/Link3_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="Link4_1">
  <inertial>
    <origin rpy="0 0 0" xyz="-1.842082955655732e-07 -0.004999749873264598 0.032486936025046875"/>
    <mass value="0.16819090005804693"/>
    <inertia ixx="0.000103" ixy="0.0" ixz="0.0" iyy="0.00011" iyz="0.0" izz="1e-05"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-1.2e-05 -0.004903 -0.325"/>
    <geometry>
      <mesh filename="package://main_description/meshes/Link4_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-1.2e-05 -0.004903 -0.325"/>
    <geometry>
      <mesh filename="package://main_description/meshes/Link4_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>



<!--  JOINTS  -->

<joint name="world_joint" type="fixed">
  <axis xyz="0 0 1"/>
	<origin xyz="0.0 0.0 0.0" rpy="0 0 0" />
	<parent link="world"/>
	<child link="base_link"/>
</joint>



<joint name="Rev1" type="continuous">
  <origin rpy="0 0 0" xyz="0.0 0.0 0.025"/>
  <parent link="base_link"/>
  <child link="Link1_1"/>
  <axis xyz="0.0 0.0 1.0"/>
  <limit effort="${strength}" lower="0.0" upper="6.283185" velocity="${speed}"/>
  <dynamics damping="${damping}" friction="0.0"/>
</joint>

<joint name="Rev2" type="revolute">
  <origin rpy="0 0 0" xyz="-2.1e-05 0.004903 0.025"/>
  <parent link="Link1_1"/>
  <child link="Link2_1"/>
  <axis xyz="0.0 -1.0 0.0"/>
  <limit effort="${strength}" lower="-1.396263" upper="1.396263" velocity="${speed}"/>
  <dynamics damping="${damping}" friction="0.0"/>
</joint>

<joint name="Rev3" type="revolute">
  <origin rpy="0 0 0" xyz="0.0 0.0 0.15"/>
  <parent link="Link2_1"/>
  <child link="Link3_1"/>
  <axis xyz="0.0 -1.0 0.0"/>
  <limit effort="${strength}" lower="-2.792527" upper="2.792527" velocity="${speed}"/>
  <dynamics damping="${damping}" friction="0.0"/>
</joint>

<joint name="Rev4" type="revolute">
  <origin rpy="0 0 0" xyz="3.3e-05 0.0 0.125"/>
  <parent link="Link3_1"/>
  <child link="Link4_1"/>
  <axis xyz="0.0 -1.0 0.0"/>
  <limit effort="${strength}" lower="-2.617994" upper="2.617994" velocity="${speed}"/>
  <dynamics damping="${damping}" friction="0.0"/>
</joint>

</robot>


<!-- 
  LINKS

<joint name="anchor" type="continuous">
  <axis xyz="0 0 1"/>
	<origin xyz="0.0 0.0 0.1" rpy="0 0 0" />
	<parent link="world"/>
	<child link="base_link"/>
  <limit effort="${strength}" lower="0" upper="6.28318" velocity="1"/>
  <dynamics damping="0.01" friction="0"/>
</joint>

<link name="base_link">
  <inertial>
    <origin rpy="0 0 0" xyz="${offset+0.05000000000001781} ${offset+0.05} -0.00045069620836176485"/>
    <mass value="1.526896640260832"/>
    <inertia ixx="0.001401" ixy="-0.0" ixz="-0.0" iyy="0.001379" iyz="-0.0" izz="0.002082"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="${0+offset} ${0+offset} 0"/>
    <geometry>
      <mesh filename="package://version1_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="${0+offset} ${0+offset} 0"/>
    <geometry>
      <mesh filename="package://version1_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="arm1_1">
  <inertial>
    <origin rpy="0 0 0" xyz="0.07999999999995012 0.0 -6.30526011055621e-08"/>
    <mass value="0.6547759683129957"/>
    <inertia ixx="6.3e-05" ixy="0.0" ixz="0.0" iyy="0.002042" iyz="-0.0" izz="0.002004"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-0.05 -0.05 -0.036536"/>
    <geometry>
      <mesh filename="package://version1_description/meshes/arm1_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-0.05 -0.05 -0.036536"/>
    <geometry>
      <mesh filename="package://version1_description/meshes/arm1_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="arm2_1">
  <inertial>
    <origin rpy="0 0 0" xyz="0.054999999999916255 0.015 -6.30526011055621e-08"/>
    <mass value="0.3898384683129956"/>
    <inertia ixx="2.9e-05" ixy="0.0" ixz="0.0" iyy="0.000669" iyz="0.0" izz="0.000655"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-0.21 -0.05 -0.036536"/>
    <geometry>
      <mesh filename="package://version1_description/meshes/arm2_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-0.21 -0.05 -0.036536"/>
    <geometry>
      <mesh filename="package://version1_description/meshes/arm2_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="arm3_1">
  <inertial>
    <origin rpy="0 0 0" xyz="0.029571233060747493 -0.014999999999999993 -6.30526011055621e-08"/>
    <mass value="0.2426509683129957"/>
    <inertia ixx="1.8e-05" ixy="0.0" ixz="-0.0" iyy="0.000191" iyz="0.0" izz="0.000182"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-0.32 -0.065 -0.036536"/>
    <geometry>
      <mesh filename="package://version1_description/meshes/arm3_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-0.32 -0.065 -0.036536"/>
    <geometry>
      <mesh filename="package://version1_description/meshes/arm3_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="HandBase_1">
  <inertial>
    <origin rpy="0 0 0" xyz="0.04972800754598111 -0.04414794632836559 -6.30526010986232e-08"/>
    <mass value="0.7824014231131748"/>
    <inertia ixx="0.001479" ixy="-0.000442" ixz="0.0" iyy="0.000967" iyz="0.0" izz="0.002301"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-0.385 -0.05 -0.036536"/>
    <geometry>
      <mesh filename="package://version1_description/meshes/HandBase_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-0.385 -0.05 -0.036536"/>
    <geometry>
      <mesh filename="package://version1_description/meshes/HandBase_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="Finger_1">
  <inertial>
    <origin rpy="0 0 0" xyz="0.03999974948858598 2.7755575615628914e-17 -6.30526011055621e-08"/>
    <mass value="0.13872095061007433"/>
    <inertia ixx="4e-06" ixy="0.0" ixz="0.0" iyy="0.000118" iyz="0.0" izz="0.000118"/>
  </inertial>
  <visual>
    <origin rpy="0 0 0" xyz="-0.415346 -0.0195 -0.036536"/>
    <geometry>
      <mesh filename="package://version1_description/meshes/Finger_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin rpy="0 0 0" xyz="-0.415346 -0.0195 -0.036536"/>
    <geometry>
      <mesh filename="package://version1_description/meshes/Finger_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<joint name="base_link" type="revolute">

  <origin rpy="0 0 0" xyz="0.0 0.0 0.036536"/>
  <parent link="base_link"/>
  <child link="arm1_1"/>
  <axis xyz="0.0 1.0 0.0"/>
  <limit effort="${strength}" lower="-3.04" upper="-0.1" velocity="${speed}"/>
  <dynamics damping="${damping}" friction="0.0"/>
</joint>

<joint name="Joint2" type="revolute">
  <origin rpy="0 0 0" xyz="0.16 0.0 0.0"/>
  <parent link="arm1_1"/>
  <child link="arm2_1"/>
  <axis xyz="0.0 1.0 0.0"/>
  <limit effort="${strength}" lower="-1.570796" upper="1.570796" velocity="${speed}"/>
  <dynamics damping="${damping}" friction="0.0"/>
</joint>

<joint name="Joint3" type="revolute">
  <origin rpy="0 0 0" xyz="0.11 0.015 0.0"/>
  <parent link="arm2_1"/>
  <child link="arm3_1"/>
  <axis xyz="0.0 1.0 0.0"/>
  <limit effort="${strength}" lower="-1.570796" upper="1.570796" velocity="${speed}"/>
  <dynamics damping="${damping}" friction="0.0"/>
</joint>

<joint name="Joint4" type="revolute">
  <origin rpy="0 0 0" xyz="0.065 -0.015 0.0"/>
  <parent link="arm3_1"/>
  <child link="HandBase_1"/>
  <axis xyz="0.0 1.0 0.0"/>
  <limit effort="${strength}" lower="-1.570796" upper="1.570796" velocity="100"/>
</joint>


<joint name="FingerSlider" type="prismatic">
  <origin rpy="0 0 0" xyz="0.030346 -0.0305 0.0"/>
  <parent link="HandBase_1"/>
  <child link="Finger_1"/>
  <axis xyz="0.0 1.0 0.0"/>
  <limit effort="${strength}" lower="-0.085" upper="0.0" velocity="100"/>
</joint> 


</robot>
-->